parquet-thrift/src/main/java/org/apache/parquet/thrift/struct/CompatibilityChecker.java

```diff
@@ -32,7 +34,7 @@
 import org.apache.parquet.thrift.struct.ThriftType.StringType;
 
 /**
- * A checker for thrift struct, returns compatibility report based on following rules:
+ * A checker for thrift struct to enforce its backward compatibility, returns compatibility report based on following rules:
  * 1. Should not add new REQUIRED field in new thrift struct. Adding optional field is OK
  * 2. Should not change field type for an existing field
```

Comment:
"4. Should not make requirement type more restrictive for a field in new thrift struct"
addresses that